src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -417,41 +417,41 @@ impl Visitor<'tcx> for Validator<'_, 'mir, 'tcx> {
             context,
             location,
         );
-        self.super_projection(place_base, proj, context, location);
```

Comment:
Change the current function to `visit_projection_elem`, this way you don't need to reroll the looping logic